### DISCUSSION TITLE
Are sublimation and phase-change heat flux from snowmelt included in latent heat flux?

### DISCUSSION THREAD

**Discussion by dmocko** [NCAR/noahmp#137] [Q&A]
I am trying to determine if the energy of sublimation and energy of the phase-change heat flux from snowmelt are included in the latent heat flux in Noah-MP.

Here is the Noah-MP-5.0 energy balance calculation:
https://github.com/NCAR/noahmp/blob/master/src/BalanceErrorCheckMod.F90#L231-L233

Sublimation (in water terms) is calculated by:
![Screenshot 2024-08-01 at 4 13 55 PM](https://github.com/user-attachments/assets/1fa18767-f3a9-4b35-a0fd-3274797de74f)

The energy of the phase-change heat flux from snowmelt is calculated by:
![Screenshot 2024-08-01 at 4 08 00 PM](https://github.com/user-attachments/assets/70420704-6718-40d7-8e70-8963fbf03aed)

Are the energy from these two terms included in the "HeatLatentGrd" term?  Or maybe also in "HeatLatentCanopy", if there is snowpack on the canopy?

One minor point, looking through the documentation I noticed that the `"H_L,grd"` term is repeated twice in the equation at the bottom of Page 240 in the v5 Technical documentation.  I believe that one of these terms should be `"H_S,grd"`.
![Screenshot 2024-08-01 at 4 20 14 PM](https://github.com/user-attachments/assets/ed8a9224-026f-4cd4-93fe-a571fd6deb3e)

**Comment by tslin2** [comment]
I think in WaterMainMod.F90, sublimation (VaporizeGrd) comes from HeatLatentGrd, so sublimation it is
https://github.com/NCAR/noahmp/blob/b0e8fa4220c2797fca40a3ac2f8631b724e8a0bf/src/WaterMainMod.F90#L110-L122

**Reply by tslin2** [reply]
For total latent heat of phase change (HeatLhTotPhsChg), directly I do not think it is but change the  snow temperature
